apis/metal3.io/v1alpha1/baremetalhost_validation.go

```diff
@@ -14,7 +15,19 @@ func (host *BareMetalHost) validateHost() []error {
 	log.Info("validate create", "name", host.Name)
 	var errs []error
 
-	if err := validateRAID(host.Spec.RAID); err != nil {
+	if err := validateRAID(host.Spec); err != nil {
```

Comment:
Instead passing whole `Spec`, we can pass just the required parameters. I think, only 1 or 2 fields are used.